tests/copp/test_copp.py

```diff
@@ -193,6 +193,25 @@ def _setup_testbed(dut, creds, ptf, test_params):
         Sets up the testbed to run the COPP tests.
     """
 
+    if dut.facts['asic_type'] in ["barefoot"]:
+        output = dut.command("docker exec syncd bash -c '[ -d /usr/local/include/nanomsg ] || echo copp'")
+
+        if output["stdout"] == "copp":
+            http_proxy = creds.get('proxy_env', {}).get('http_proxy', '')
+            https_proxy = creds.get('proxy_env', {}).get('https_proxy', '')
+
+            cmd = '''docker exec -e http_proxy={} -e https_proxy={} syncd bash -c " \
+                    apt-get update \
+                    && apt-get install -y python-pip build-essential libssl-dev python-dev wget cmake \
+                    && wget https://github.com/nanomsg/nanomsg/archive/1.0.0.tar.gz \
+                    && tar xvfz 1.0.0.tar.gz && cd nanomsg-1.0.0 \
+                    && mkdir -p build && cmake . && make install && ldconfig && cd .. && rm -fr nanomsg-1.0.0 \
+                    && rm -f 1.0.0.tar.gz && pip install cffi==1.7.0 && pip install --upgrade cffi==1.7.0 && pip install nnpy \
+                    && mkdir -p /opt && cd /opt && wget https://raw.githubusercontent.com/p4lang/ptf/master/ptf_nn/ptf_nn_agent.py \
+                    && mkdir ptf && cd ptf && wget https://raw.githubusercontent.com/p4lang/ptf/master/src/ptf/afpacket.py && touch __init__.py \
+                    " '''.format(http_proxy, https_proxy)
+            dut.command(cmd)
```

Comment:
I would also consider making this an option like `copp_swap_syncd` rather than hardcoding it based on the ASIC type.

Reply:
@daall , @OleksandrKozodoi , since to run CoPP we need `nanomsg` anyway and this is not platform specific, I believe we don't need either extra option or platform check. In case syncd has been swapped with syncd-rpc, the `nanomsg` package is already there and this code won't be executed.

Please see: https://github.com/Azure/sonic-buildimage/search?q=nanomsg&unscoped_q=nanomsg

Reply:
That does seem quite a bit cleaner. It makes sense to me to add that to the beginning of `configure_syncd` before the template step.

Reply:
Also can we give it a slightly different name like `install_nano` or `install_nn_agent` so that it's more clear what the difference is between this method and `configure_syncd`? :)

Reply:
Done. Thanks

Reply:
> Also can we give it a slightly different name like `install_nano` or `install_nn_agent` so that it's more clear what the difference is between this method and `configure_syncd`? :)

Danny, could you please review my PR? Thanks :)